### PR TITLE
mgr/dashboard: Provide the name of the object being deleted

### PR DESCRIPTION
**Pull request by ricardoasmarques**
Screenshot from 2019-10-01 12-21-23

Fixes: https://tracker.ceph.com/issues/41687

Signed-off-by: Ricardo Marques <rimarques@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/critical-confirmation-modal/critical-confirmation-modal.component.html [hunk 1]
```diff
@@ -13,7 +13,8 @@
         <ng-container *ngTemplateOutlet="bodyTemplate; context: bodyContext"></ng-container>
         <div class="question">
           <p i18n>Are you sure that you want to
-            {{ actionDescription | lowercase }} the selected {{ itemDescription }}?</p>
+            {{ actionDescription | lowercase }}
+            <span *ngIf="!itemName">the selected {{ itemDescription }}</span><strong *ngIf="itemName">{{ itemName }}</strong>?</p>
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/components/critical-confirmation-modal/critical-confirmation-modal.component.html]: Minor stuff: `ngIf/else`?

```suggestion
            <strong *ngIf="itemName; else noName">{{ itemName }}</strong><span #noName>the selected {{ itemDescription }}</span>?</p>
```

**Comment by votdev** [comment]
Have you thought about how to support lists that can do multi-selection? Maybe itemName can be a callback function in that case to process the selected items and return the text to be displayed.

**Comment by ricardoasmarques** [comment]
> Have you thought about how to support lists that can do multi-selection? Maybe itemName can be a callback function in that case to process the selected items and return the text to be displayed.

I think that in the future `itemName` can be a `string` or a `string list`, and for the case it is a list I imagine something like:

```
Are you sure that you want to delete the following items?
- myppol1
- mypool2
```

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/critical-confirmation-modal/critical-confirmation-modal.component.html (hunk 1 above)
Comment (ricardoasmarques) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/components/critical-confirmation-modal/critical-confirmation-modal.component.html]: Done

**Comment by ricardoasmarques** [comment]
> > Have you thought about how to support lists that can do multi-selection? Maybe itemName can be a callback function in that case to process the selected items and return the text to be displayed.
> 
> I think that in the future `itemName` can be a `string` or a `string list`, and for the case it is a list I imagine something like:
> 
> ```
> Are you sure that you want to delete the following items?
> - mypool1
> - mypool2
> ```

@votdev This PR is now adapted to support multi-selection

**Comment by ricardoasmarques** [comment]
jenkins test docs

**Comment by callithea** [comment]
I've tested it locally in combination with PR #30495 but when selecting several items (e.g. RGW users) it only shows the first selected item name in the dialog - not sure if it's reproducible for anyone else.

**Comment by ricardoasmarques** [comment]
> I've tested it locally in combination with PR #30495 but when selecting several items (e.g. RGW users) it only shows the first selected item name in the dialog - not sure if it's reproducible for anyone else.

@callithea Thanks for testing, that was indeed a bug. It should be fixed now.

**Comment by ricardoasmarques** [comment]
jenkins test make check arm64

**Comment by ricardoasmarques** [comment]
jenkins test make check arm64

**Pull request merged**